osquery/tables/system/windows/windows_search.cpp

```diff
@@ -0,0 +1,254 @@
+// osquery/tables/system/windows/windows_search.cpp
+
+/**
+ * Copyright (c) 2014-present, The osquery authors
+ *
+ * This source code is licensed as defined by the LICENSE file found in the
+ * root directory of this source tree.
+ *
+ * SPDX-License-Identifier: (Apache-2.0 OR GPL-2.0-only)
+ */
+
+#include <windows.h>
+#include <atldbcli.h>
+#include <sstream>
+#include <vector>
+#include <comutil.h>
+#include <strsafe.h>
+#include <codecvt>
+#include <osquery/core/core.h>
+#include <osquery/core/tables.h>
+#include <osquery/logger/logger.h>
+#include <osquery/utils/conversions/windows/strings.h>
+
+#pragma comment(lib, "comsuppw.lib")
+
+namespace osquery {
+namespace tables {
+
+void writeDate(double date, std::wstringstream& wss)
+{
+    SYSTEMTIME sysTime;
+    BOOL ok = VariantTimeToSystemTime(date, &sysTime);
+    if (ok)
+    {
+        SYSTEMTIME localTime;
+        ok = SystemTimeToTzSpecificLocalTime(NULL, &sysTime, &localTime);
+        if (ok)
+        {
+            WCHAR szBuffer[100];
+            ok = GetDateFormat(LOCALE_USER_DEFAULT, 0, &localTime, NULL, szBuffer, ARRAYSIZE(szBuffer) / sizeof(WCHAR));
```

Comment:
Could you explain this line to me, please?
`ARRAYSIZE` macro usually returns size of the array , e.g. it can tell how many elements in the array.
Why it is divided by the element size once more?

Reply:
I'm not able to explain that, it's some copy pasta I took from some windows examples. However, looking at some tables, it seems that unix time is the standard for osquery, so now the dates are returned as unix time stamps.